vortex-filtering/include/vortex_filtering/filters/ukf.hpp

```diff
@@ -0,0 +1,309 @@
+  {
+    Mat_xx P = x_est.cov();
+    Mat_vv Q = dyn_mod->Q_d(dt, x_est.mean());
+    Mat_ww R = sens_mod->R(x_est.mean());
+    // Make augmented covariance matrix
+    Mat_aa P_a;
+    // clang-format off
+    P_a << P	           , Mat_xv::Zero() , Mat_xw::Zero(),
+           Mat_vx::Zero(), Q              , Mat_vw::Zero(),
+           Mat_wx::Zero(), Mat_wv::Zero() , R;
+    // clang-format on
+    Mat_aa sqrt_P_a = P_a.llt().matrixLLT();
+
+    // Make augmented state vector
+    Vec_a x_a;
+    x_a << x_est.mean(), Vec_v::Zero(), Vec_w::Zero();
+
+    // Calculate sigma points using the symmetric sigma point set
+    Mat_a2ap1 sigma_points;
+    sigma_points.col(0) = x_a;
+    for (int i = 1; i <= N_DIM_a; i++) {
+      sigma_points.col(i)           = x_a + GAMMA * sqrt_P_a.col(i - 1);
+      sigma_points.col(i + N_DIM_a) = x_a - GAMMA * sqrt_P_a.col(i - 1);
+    }
+    return sigma_points;
+  }
+
+  /** Propagate sigma points through f
+   * @param dyn_mod Dynamic model
+   * @param dt Time step
+   * @param sigma_points Mat_a2ap1 Sigma points
+   * @param u Vec_u Control input
+   * @return Mat_x2ap1 sigma_x_pred
+   */
+  static Mat_x2ap1 propagate_sigma_points_f(const DynModIPtr &dyn_mod, double dt, const Mat_a2ap1 &sigma_points, const Vec_u &u = Vec_u::Zero())
+  {
+    Eigen::Matrix<double, N_DIM_x, 2 * N_DIM_a + 1> sigma_x_pred;
+    for (int i = 0; i < 2 * N_DIM_a + 1; i++) {
+      Vec_x x_i           = sigma_points.template block<N_DIM_x, 1>(0, i);
+      Vec_v v_i           = sigma_points.template block<N_DIM_v, 1>(N_DIM_x, i);
+      sigma_x_pred.col(i) = dyn_mod->f_d(dt, x_i, u, v_i);
+    }
+    return sigma_x_pred;
+  }
+
+  /** Propagate sigma points through h
+   * @param sens_mod Sensor model
+   * @param sigma_points Mat_a2ap1 Sigma points
+   * @return Mat_z2ap1 sigma_z_pred
+   */
+  static Mat_z2ap1 propagate_sigma_points_h(const SensModIPtr &sens_mod, const Mat_a2ap1 &sigma_points)
+  {
+    Mat_z2ap1 sigma_z_pred;
+    for (int i = 0; i < 2 * N_DIM_a + 1; i++) {
+      Vec_x x_i           = sigma_points.template block<N_DIM_x, 1>(0, i);
+      Vec_w w_i           = sigma_points.template block<N_DIM_w, 1>(N_DIM_x + N_DIM_v, i);
+      sigma_z_pred.col(i) = sens_mod->h(x_i, w_i);
+    }
+    return sigma_z_pred;
+  }
+
+  /** Estimate gaussian from sigma points using the unscented transform
+   * @param sigma_points Mat_n2ap1 Sigma points
+   * @tparam n_dims Dimension of the gaussian
+   * @return prob::MultiVarGauss<n_dims>
+   * @note This function is templated to allow for different dimensions of the gaussian
+   */
+  template <int n_dims> static prob::MultiVarGauss<n_dims> estimate_gaussian(const Eigen::Matrix<double, n_dims, 2 * N_DIM_a + 1> &sigma_points)
+  {
+    using Vec_n  = Eigen::Vector<double, n_dims>;
+    using Mat_nn = Eigen::Matrix<double, n_dims, n_dims>;
+
+    // Predicted State Estimate x_k-
+    Vec_n mean = Vec_n::Zero();
+    for (int i = 0; i < 2 * N_DIM_a + 1; i++) {
```

Comment:
Consider making 2 * N_DIM_a + 1 a static constexpr (same goes for the other ones)